### ISSUE TITLE
Observable metric exception shutdown 

### ISSUE THREAD

**Issue by RafaelCavazza** [open-telemetry/opentelemetry-java#3041]
When defining an observable metric if the updater throws a  RuntimeException the MetricExporter stops without any log or warning.

I believe this behavior is caused because the 'ScheduledThreadPoolExecutor' suppress all subsequent executions when encounters an exception.

This is a unwanted behavior since one metric can shutdown all others without notice.

Ex.:
````
meter.doubleValueObserverBuilder("sample")
                .setUpdater(c -> {
                    throw new RuntimeException("There was an unexpected error!");
                }).build();
````

Version:
opentelemetry-api-metrics - 1.0.1-alpha
opentelemetry-sdk-metrics - 1.0.1-alpha

Environment:
openjdk version "11.0.10" 2021-01-19
OpenJDK Runtime Environment GraalVM CE 21.0.0.2 (build 11.0.10+8-jvmci-21.0-b06)
OpenJDK 64-Bit Server VM GraalVM CE 21.0.0.2 (build 11.0.10+8-jvmci-21.0-b06, mixed mode, sharing)
OS: Mac

**Comment by Oberon00** [comment]
@RafaelCavazza If you are able to verify with a version from the main branch, it would be cool if you could check if that issue is resolved with #3044.

**Comment by RafaelCavazza** [comment]
Hi @Oberon00 !

It is, thanks!
